include/itkWaveletCoeffsInverseFFTImageFilter.hxx

```diff
@@ -0,0 +1,133 @@
+template< typename TImageType, 
+  typename TWaveletFunction > 
+WaveletCoeffsInverseFFTImageFilter< TImageType, TWaveletFunction > 
+:: WaveletCoeffsInverseFFTImageFilter()
+{
+  m_Levels = 4;
+  m_HighPassSubBands = 3;
+}
+
+template< typename TImageType, 
+  typename TWaveletFunction > 
+void 
+WaveletCoeffsInverseFFTImageFilter< TImageType, TWaveletFunction > 
+:: GenerateData()
+{
+  m_FFTPadFilter = FFTPadType::New();
+  m_ZeroDCFilter = ZeroDCType::New();
+  m_ForwardFFTFilter = FFTForwardType::New();
+  m_ForwardWaveletFilter = ForwardWaveletType::New();
+
+  typename ImageType::Pointer input = ImageType::New();
+  input-> Graft(const_cast<ImageType*> ( this-> GetInput() ));
+  m_FFTPadFilter-> SetInput( input );
+
+  m_ZeroDCFilter-> SetInput( m_FFTPadFilter-> GetOutput() );
+  m_ForwardFFTFilter-> SetInput( m_ZeroDCFilter-> GetOutput() );
+  m_ForwardWaveletFilter-> SetInput( m_ForwardFFTFilter-> GetOutput() );
+
+  m_ForwardWaveletFilter-> SetHighPassSubBands( this-> m_HighPassSubBands );
+  m_ForwardWaveletFilter-> SetLevels( this-> m_Levels );
+
+  m_InverseFFTFilter = InverseFFTType::New();
+  m_ChangeInformationFilter = ChangeInformationType::New();
+  m_CastFloatFilter = CastFloatType::New();
+
+  unsigned int k = (this-> m_Levels * this-> m_HighPassSubBands);
+  this-> SetNumberOfRequiredOutputs(k + 1);
+  this-> Modified();
+  for ( unsigned int i = 0; i < k + 1; ++i )
+    {
+    this->SetNthOutput(i, this->MakeOutput(i));
+    }
+
+  for ( unsigned int i = 0; i < k + 1; ++i )
+    {
+    m_InverseFFTFilter-> SetInput( m_ForwardWaveletFilter-> GetOutput(i) ); // This API is strange.
+    m_ChangeInformationFilter-> SetInput( m_InverseFFTFilter-> GetOutput() );
+
+    m_ChangeInformationFilter-> SetReferenceImage(m_FFTPadFilter-> GetOutput());
+    m_ChangeInformationFilter-> UseReferenceImageOn();
+    m_ChangeInformationFilter-> ChangeAll();
+
+    m_CastFloatFilter-> SetInput( m_ChangeInformationFilter-> GetOutput() );
+
+    typename ImageType::Pointer modifiedWavelets;
+    // TODO Warning: InPlace here deletes buffered region of input.
+
+    m_CastFloatFilter-> InPlaceOff();
+    m_CastFloatFilter-> Update();
+    modifiedWavelets = m_CastFloatFilter-> GetOutput();
+
+    // We are going to disconnect the object output of the filter pointer
+    // So m_filter-> GetOutput()-> DisconnectPipeline();
+    modifiedWavelets-> DisconnectPipeline();
+
+    // this-> SetNthOutput(i, this-> MakeOutput(modifiedWavelets));
+    // this-> SetNthOutput(i, modifiedWavelets); // ==> This is the working API; but looks like we don't need this.
+
+    this-> GraftNthOutput(i, modifiedWavelets );
+    }
+}
+
+template< typename TImageType, 
+  typename TWaveletFunction > 
+  void 
+  WaveletCoeffsInverseFFTImageFilter< TImageType, TWaveletFunction > 
+  :: PrintSelf( std::ostream & os, Indent indent ) const
+{
+  Superclass::PrintSelf(os, indent);
+  os  << indent
+      << "==============================================================\n"
```

Comment:
For consistent *PrintSelf* style, the *=====* should be removed and there should be a single parameter per line.